src/Advection/tracer_advection_operators.jl

```diff
@@ -1,6 +1,34 @@
 using Oceananigans.Operators: Vᶜᶜᶜ
 using Oceananigans.Fields: ZeroField
 
+struct TracerAdvection{N, FT, A, B, C} <: AbstractAdvectionScheme{N, FT}
+    x :: A
+    y :: B
+    z :: C
+
+    TracerAdvection{N, FT}(x::A, y::B, z::C) where {N, FT, A, B, C} = new{N, FT, A, B, C}(x, y, z)
+end
+
+"""
+    function TracerAdvection(; x, y, z)
+
+Builds a `TracerAdvection` type with reconstruction schemes in `x`, `y`, and `z`.
+"""
+function TracerAdvection(; x, y, z)
+    Nx = required_halo_size(x)
+    Ny = required_halo_size(y)
+    Nz = required_halo_size(z)
+
+    FT = eltype(x)
+
+    return TracerAdvection{max(Nx, Ny, Nz), FT}(x, y, z)
+end
+
+Adapt.adapt_structure(to, scheme::TracerAdvection{N, FT}) where {N, FT} =
+    TracerAdvection{N, FT}(Adapt.adapt(to, scheme.x),
+                           Adapt.adapt(to, scheme.y),
+                           Adapt.adapt(to, scheme.z))
```

Comment:
What does this stuff have to do with CATKE performance optimization?

Reply:
that is code from another PR #3404 we have to merge before this one